src/engine/Join.cpp

```diff
@@ -762,77 +763,56 @@ void Join::hashJoin(const IdTable& dynA, size_t jc1, const IdTable& dynB,
 
   IdTableStatic<OUT_WIDTH> result = dynRes->moveToStatic<OUT_WIDTH>();
 
-  // Cannot just switch l1 and l2 around because the order of
+  // Puts the rows of the given table into a hash map, with the value of
+  // the join column of a row as the key, and returns the hash map.
+  auto putIntoHashMap = [] (const auto table, const size_t jc, const auto WIDTH) {
```

Comment:
Call this function `idTableToHashMap`, and make the type of the `table` a `const&` and make its type explicitly named:
```
auto putIntoHashMap = []<typename Table> (const Table& table, const size_t jc){ 
```
(I removed the WIDTH as we can infer it from the `Table`.

Reply:
Small problem with infering it from the table: This value is not always known at compile time, so you can't use it as a compile time variable, because the compiler will complain.

More specific, if the `IdTableImpl` has 0 columns, the amount of columns is only known at runtime. Which makes `.cols()` (Returning the amount of columns of the table) non `constexpr`, even though for all other cases this information is known at compile time. So the compiler doesn't want it to be used for template arguments.

Reply:
I found an alternate path, by using parameter inference `[]<int Width> (const IdTableView<Width>& table, const size_t jc)` instead of trying to get the width from `table`, which is only known at compile time. Is this okay, or do you see a better way of doing that?

See [9041109](https://github.com/ad-freiburg/qlever/pull/821/commits/9041109119903a1b1c9154d9a924632de42342b8).

Reply:
Nevermind, it works when combined with this [change](https://github.com/ad-freiburg/qlever/pull/821#discussion_r1023161914). You just have to get rid of the whole idea of using `WIDTH` in the first place, which I didn't think of.

Reply:
Done. See [88e172c](https://github.com/ad-freiburg/qlever/pull/821/commits/88e172c9caec3fdc591de766d50861acdaad4a94).